### PR TITLE
added opt-in no regex

### PR DESCRIPTION
**Pull request by ncave**
Not a large perf difference, but it can be a little bit faster on big projects. Opt-in with compilation define.

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
Performance improvements are always welcome! I'm OK with just replacing the Regex code to avoid the need of using the compilation define, what do you think? Or do you prefer to keep the old code in cases there are scenarios the new code cannot handle?

**Comment by ncave** [comment]
@alfonsogarciacaro No, I was just trying to avoid controversy. I'll take it out entirely.

**Comment by alfonsogarciacaro** [comment]
Gimme my Regex back! ...just kidding, you know in F# controversies only happen around the `fun` keyword ...well, not only :wink:

Thanks!

File: src/dotnet/Fable.Compiler/Replacements.fs [hunk 1]
```diff
@@ -2537,8 +2536,9 @@ let checkLiteral com fileName range (value: obj) (typ: Fable.Type) =
     match typ with
     | Fable.Enum("System.Text.RegularExpressions.RegexOptions") ->
         match value with
-        | :? int as i when i = int RegexOptions.IgnoreCase
-                        || i = int RegexOptions.Multiline
-                        || i = int RegexOptions.ECMAScript -> ()
+        | :? int as i when i = 0x0001 // "i" - RegexOptions.IgnoreCase
+                        || i = 0x0002 // "m" - RegexOptions.Multiline
+                        || i = 0x0100 // "e" - RegexOptions.ECMAScript
```
Comment (inosik) [inline src/dotnet/Fable.Compiler/Replacements.fs]: This is not necessary. The IL code is exactly the same.

Comment (ncave) [reply src/dotnet/Fable.Compiler/Replacements.fs]: @inosik The point is to not use anything from the regex namespace (in this case, RegexOptions) to lower the API surface.

**Pull request merged**